### PR TITLE
GG-25424 Non-comparable keys for eviction policy cause failure handle…

### PR DESCRIPTION
**Pull request by ingvard**
… and node shutdown

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/eviction/EvictionPolicyFailureHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+import org.apache.ignite.transactions.Transaction;
+import org.junit.Test;
+
+import static org.apache.ignite.transactions.TransactionConcurrency.PESSIMISTIC;
+import static org.apache.ignite.transactions.TransactionIsolation.REPEATABLE_READ;
+
+/**
+ * Sorted eviction policy tests.
+ */
+public class EvictionPolicyFailureHandlerTest extends GridCommonAbstractTest {
```
Comment (sanpwc) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/eviction/EvictionPolicyFailureHandlerTest.java]: Seems that given test isn't added to a suite.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/eviction/EvictionPolicyFailureHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        }
+
+        for (int i = 0; i < 1000; i++) {
+            double d = (double)i;
+
+            if (affinity.isPrimary(node1.localNode(), d))
+                cache.put(d, 1);
+        }
+
+        assertFalse(nodeFailure.get());
```
Comment (sanpwc) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/eviction/EvictionPolicyFailureHandlerTest.java]: Could you please add extra check in order to validate that all "double-typed" entries were evicted from on-heap?

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/eviction/EvictionPolicyFailureHandlerTest.java (hunk 1 above)
Comment (ingvard) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/eviction/EvictionPolicyFailureHandlerTest.java]: It's true, I added it.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/eviction/EvictionPolicyFailureHandlerTest.java (hunk 2 above)
Comment (ingvard) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/eviction/EvictionPolicyFailureHandlerTest.java]: Yes, I added it.

**Pull request closed without merging**